lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -2835,6 +2854,102 @@ OpFoldResult AtenDivIntOp::fold(FoldAdaptor adaptor) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// AtenIndexSelectOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenIndexSelectOp::fold(FoldAdaptor adaptor) {
+  auto self = getSelf();
+  auto index = getIndex();
+  auto selfTy = dyn_cast_or_null<ValueTensorType>(self.getType());
+  auto indexTy = dyn_cast_or_null<ValueTensorType>(index.getType());
+  auto resultTy = dyn_cast_or_null<ValueTensorType>(getType());
+  if (!selfTy || !indexTy || !resultTy)
+    return nullptr;
+
+  if (!selfTy.hasSizes() || !indexTy.hasSizes() || !resultTy.hasSizes())
+    return nullptr;
+
+  if (!selfTy.hasDtype() || !indexTy.hasDtype() || !resultTy.hasDtype())
+    return nullptr;
+
+  auto selfSizes = selfTy.getSizes();
+  auto indexSizes = indexTy.getSizes();
+  auto resultSizes = resultTy.getSizes();
+
+  if (selfTy.getDtype() != resultTy.getDtype())
```

Comment:
You might as well `||` of these checks together so we save a few lines.

Reply:
done

Reply:
Did you push these changes? I am not seeing the if statements merged.

Reply:
My bad!  Got confused by multiple blocks of ifs.  We had a chat about the new fix offline, so resolving now.  Thanks for all your help!